### PR TITLE
Improve service history enrollment processing queue from hmis

### PR DESCRIPTION
**Pull request by gigxz**
## Description

Fix issue where processing jobs get piled up from HMIS

## Type of change
- [x] Bug fix

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: drivers/hmis/app/models/hmis/hud/concerns/service_history_queuer.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    #
+    # If job is already queued, do nothing
+    # If job is currently running, queue it to run 5 minutes from now.
+    # Otherwise, queue the job.
+    def queue_service_history_processing!
+      handlers = ['GrdaWarehouse::Tasks::ServiceHistory::Enrollment', 'batch_process_unprocessed!']
+      return if Delayed::Job.queued?(handlers)
+
+      currently_running = Delayed::Job.running?(handlers)
+      run_at = currently_running ? 5.minutes.from_now : nil
```
Comment (eanders) [inline drivers/hmis/app/models/hmis/hud/concerns/service_history_queuer.rb:21]: maybe?
```suggestion
      run_at = currently_running ? 5.minutes.from_now : Time.current
```

Comment (gigxz) [reply drivers/hmis/app/models/hmis/hud/concerns/service_history_queuer.rb:21]: What happens when that time is slightly in the past, though? I wasn't sure, so I figured it's safer to give it the default behavior.

**Comment by gigxz** [comment]
Alright, ready for review! Things are working as expected locally.

File: drivers/hmis/app/models/hmis/hud/concerns/service_history_queuer.rb (hunk 1 above)
Comment (eanders) [reply drivers/hmis/app/models/hmis/hud/concerns/service_history_queuer.rb:21]: default behavior is good.  If the time is in the past it assumes it should be run.  I just was unclear if passing a nil was the same as not passing the argument in this case.

**Review by eanders — APPROVED** [review]

File: drivers/hmis/app/models/hmis/hud/concerns/service_history_queuer.rb (hunk 1 above)
Comment (gigxz) [reply drivers/hmis/app/models/hmis/hud/concerns/service_history_queuer.rb:21]: gotcha. I looked at the code and they should both be the same

**Pull request merged**